### PR TITLE
scripts: build: parse_syscalls: Fix duplicate paths on windows

### LINKED ISSUE #59340: Encounter build error caused by redefinition of declarations on Windows
**Describe the bug**
A build error caused by redefinitions of `c macro` and `function declarations`.

**To Reproduce**
1. cd samples\hello_world
2. set BOARD=npcx9m6f_evb
3. west build -p always

**Expected behavior**
build successfully.

**Logs and console output**
Please refer attached [build_error.zip](https://github.com/zephyrproject-rtos/zephyr/files/11784865/build_error.zip). We found lots of  redefined warnings/errors in `syscall_list.h` and `kernel.h` generated by `gen_syscalls.py`
```
In file included from C:/Users/TomChang/Desktop/zephyrproject/zephyr/include/zephyr/syscall.h:11,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/include/zephyr/kernel_includes.h:31,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/include/zephyr/kernel.h:17,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/arch/arm/core/offsets/offsets_aarch32.c:28,
                 from C:/Users/TomChang/Desktop/zephyrproject/zephyr/arch/arm/core/offsets/offsets.c:9:
C:/Users/TomChang/Desktop/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/syscall_list.h:215: warning: "K_SYSCALL_CLOCK_GETTIME" redefined
  215 | #define K_SYSCALL_CLOCK_GETTIME 206
      | 
C:/Users/TomChang/Desktop/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/syscall_list.h:6: note: this is the location of the previous definition
    6 | #define K_SYSCALL_CLOCK_GETTIME 0
```

**Environment (please complete the following information):**
 - OS:  Windows
 - Toolchain - Zephyr SDK, 15.1
 - Commit SHA: 32fae07f299aa326

### PR DESCRIPTION
**Pull request by nordicjm**
Fixes an issue on windows where there would be many permutations
    of paths with different folder separators by converting to
    posix-style path separators.
    
    Issue was introduced by
    https://github.com/zephyrproject-rtos/zephyr/pull/58351

Fixes #59340

**Needs verification to ensure this is the proper fix.**

### CODE REVIEW

**Review by MulinChao — DISMISSED** [review]
No more build errors on Windows after applying this.

File: scripts/build/parse_syscalls.py [hunk 1]
```diff
@@ -101,6 +102,11 @@ def analyze_headers(include_dir, scan_dir, file_list):
                                                    'common.h'))):
                     continue
 
+                # On windows, change to posix style paths to avoid duplicate
+                # entries with differing folder separators.
+                if os.name == 'nt':
+                    path = PureWindowsPath(path).as_posix()
```
Comment (tejlmand) [inline scripts/build/parse_syscalls.py:108]: any reason this needs to be tied to the os.

Why not just use PurePath and have identical code across all platforms ?
```suggestion
                PurePath(os.path.normpath(path)).as_posix())
```

**Review by tejlmand — COMMENTED** [review]
small observation to be explained before approval.

**Review by nashif — DISMISSED** [review]

File: scripts/build/parse_syscalls.py [hunk 2]
```diff
@@ -29,6 +29,7 @@
 import argparse
 import os
 import json
+from pathlib import PureWindowsPath
```
[the remark this answers is not in the document]
Comment (nordicjm) [reply scripts/build/parse_syscalls.py]: Fixed

File: scripts/build/parse_syscalls.py (hunk 1 above)
Comment (nordicjm) [reply scripts/build/parse_syscalls.py]: Fixed

**Review by tejlmand — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**